### DISCUSSION TITLE
Where did `gix::worktree::checkout` go in `gix` 0.52?

### DISCUSSION THREAD

**Discussion by Shnatsel** [GitoxideLabs/gitoxide#1000] [Q&A] [answered]
`cargo-audit` uses [`gix::worktree::checkout`](https://docs.rs/gix/0.51.0/gix/worktree/fn.checkout.html) from `gix` crate v0.51.x, [here](https://github.com/rustsec/rustsec/blob/8280d86aa5d61194a2816e2d4a4dae21b8a40f62/rustsec/src/repository/git/commit.rs#L144-L162).

I am in the process of [upgrading](https://github.com/rustsec/rustsec/pull/957) to gix 0.52, and I cannot find a checkout function in 0.52. I've [searched](https://docs.rs/gix/latest/gix/index.html?search=checkout) docs.rs for "checkout" but there seemed to be no matching functions in 0.52. How can I perform checkouts in the latest release?

**Comment by niklaswimmer** [comment]
It seems like this functionality was split into a new crate: https://docs.rs/gix-worktree-state/latest/gix_worktree_state/

**Comment by Byron** [comment, marked as answer]
Sorry for the hassle. This is in preparation for compile time improvements, and now the method is available in `gix::worktree::state::checkout()`.

**Reply by Shnatsel** [reply]
That worked. Thanks!

For some reason this method [does not come up](https://docs.rs/gix/0.52.0/gix/index.html?search=checkout) in search on docs.rs, you might want to look into that.

**Reply by Byron** [reply]
It seems that `rustdoc` doesn't index crates that are re-exported via `pub use`, and maybe that's good as it's a plumbing crate with a lower-level interface. At some point all of these should be moved behind an `unstable` flag if they are pre-1.0 as well.

In any case, I am glad it works now.

**Reply by Shnatsel** [reply]
The 0.51 function is indexed just fine: https://docs.rs/gix/0.51.0/gix/index.html?search=checkout

**Reply by Byron** [reply]
Can it be there is a difference between a top-level reexport (in `gix`) like this and one that is in `gix::worktree`? Apparently it is, and I wonder if it's intentional. I can click through to the `checkout` function too, it's just not listed in the search if re-exported from a sub-module.
